viral/multiple_sessions.py

```diff
@@ -262,7 +323,243 @@ def get_chance_level(mice: List[MouseSummary], window: int) -> List[float]:
+    plt.show()
+
+
+def plot_trial_time_summaries(
+    mice: List[MouseSummary], condition: str = "pre_reversal"
+):
+    trial_time_dict = {}
+    for mouse in mice:
+        pre_reversal = [
+            session
+            for session in mouse.sessions
+            if "reversal" not in session.name.lower()
+        ]
+        post_reversal = [
+            session for session in mouse.sessions if "reversal" in session.name.lower()
+        ]
+        assert len(pre_reversal) + len(post_reversal) == len(mouse.sessions)
+
+        # overall speed
+        trial_time_dict[mouse.name] = {
+            "pre_reversal_unrewarded": trial_time(
+                flatten_sessions(pre_reversal), False
+            ),
+            "post_reversal_unrewarded": trial_time(
+                flatten_sessions(post_reversal), False
+            ),
+            "pre_reversal_rewarded": trial_time(flatten_sessions(pre_reversal), True),
+            "post_reversal_rewarded": trial_time(flatten_sessions(post_reversal), True),
+        }
+
+    to_plot = {
+        "NLGF\nRewarded": [
+            data[f"{condition}_rewarded"]
+            for mouse, data in trial_time_dict.items()
+            if data[f"{condition}_rewarded"] > 1
+            if get_genotype(mouse) == "NLGF"
+        ],
+        "NLGF\nUnrewarded": [
+            data[f"{condition}_unrewarded"]
+            for mouse, data in trial_time_dict.items()
+            if data[f"{condition}_unrewarded"] > 1
+            if get_genotype(mouse) == "NLGF"
+        ],
+        "Oligo-Bace1\nKnockout\nRewarded": [
+            data[f"{condition}_rewarded"]
+            for mouse, data in trial_time_dict.items()
+            if data[f"{condition}_rewarded"] > 1
+            if get_genotype(mouse) == "Oligo-BACE1-KO"
+        ],
+        "Oligo-Bace1\nKnockout\nUnrewarded": [
+            data[f"{condition}_unrewarded"]
+            for mouse, data in trial_time_dict.items()
+            if data[f"{condition}_unrewarded"] > 1
+            if get_genotype(mouse) == "Oligo-BACE1-KO"
+        ],
+        "WT\nRewarded": [
+            data[f"{condition}_rewarded"]
+            for mouse, data in trial_time_dict.items()
+            if data[f"{condition}_rewarded"] > 1
+            if get_genotype(mouse) == "WT"
+        ],
+        "WT\nUnrewarded": [
+            data[f"{condition}_unrewarded"]
+            for mouse, data in trial_time_dict.items()
+            if data[f"{condition}_unrewarded"] > 1
+            if get_genotype(mouse) == "WT"
+        ],
+    }
+    plt.ylabel(f"Trial time (s)")
+    plt.title(condition.replace("_", " ").capitalize())
+
+    sns.boxplot(to_plot, showfliers=False)
+    sns.stripplot(to_plot, edgecolor="black", linewidth=1)
+    plt.tight_layout()
+    plt.savefig(HERE.parent / "plots" / f"behaviour-summaries-trial-time-{condition}")
+    plt.show()
+
+
+def plot_num_trials_summaries(
+    mice: List[MouseSummary], condition: str = "pre_reversal"
+):
```

Comment:
I think there's a lot of repeated code between these plotting functions that can be refactored out into its own function